### DISCUSSION TITLE
I need help with ReSharper command line tools

### DISCUSSION THREAD

**Discussion by AlaiyaNetwork** [alirezanet/Husky.Net#98] [Q&A] [answered]
I add this task
```
{
    "name": "Clean up code",
    "group": "clean-up-code",
    "command": "cmd",
    "pathMode": "relative",
    "args": [
        "/c",
        "dotnet",
        "jb",
        "cleanupcode",
        "My.sln",
        "--profile=Teams: Clean up code",
        "--settings=My.sln.DotSettings",
        "--include=${stages-files}"
    ]
}
```

And variables:
```
{
    "name": "stages-files",
    "command": "cmd",
    "args": [
        "/c",
        "git diff --staged --name-only --no-ext-diff --diff-filter=AM | tr '\\n' ';' | sed 's/;$//'"
    ]
}
```

If I commited  none *.cs files: 

```
[Husky]  Preparing task 'Clean up code'
[Husky]  Executing task 'Clean up code' ...
JetBrains Cleanup Code 2023.3 RC 1
Running on x64 OS in x64 architecture, .NET 8.0.0 under Microsoft Windows 10.0.22631
No items were found to cleanup.
  ❌ Task 'Clean up code' failed in 16 864ms
husky - pre-commit hook exited with code 1 (error)
```

Maybe someone has already had this problem and can suggest a good solution?

**Comment by alirezanet** [comment]
Hi @AlaiyaNetwork,
I would suggest to use the `Include` or `Exclude` attributes to filter it out, [More info](https://alirezanet.github.io/Husky.Net/guide/task-configuration.html#glob-patterns)

```
{
    "name": "Clean up code",
    "group": "clean-up-code",
    "command": "cmd",
    "pathMode": "relative",
    "include": ["**/*.cs", "**/*.vb"], // something like this should work as expected
    "args": [
        "/c",
        "dotnet",
        "jb",
        "cleanupcode",
        "My.sln",
        "--profile=Teams: Clean up code",
        "--settings=My.sln.DotSettings",
        "--include=${stages-files}"
    ]
}
```

**Reply by AlaiyaNetwork** [reply]
Yes. I've already tried it. But Include/Exclude doesn't work

**Reply by alirezanet** [reply]
Ow, I see because you're not using `${staged}` in your task, 
Can I ask why you defined a custom variable for staged files?

**Reply by AlaiyaNetwork** [reply]
To specify multiple paths or wildcards, separate them with the semicolon ; or use the --include/--exclude parameters multiple times.

This is from the ReSharper Command Line Tools documentation

**Reply by AlaiyaNetwork** [reply]
```
{
    "name": "Clean up code",
    "group": "clean-up-code",
    "command": "cmd",
    "pathMode": "relative",
    "include": ["**/*.cs", "**/*.vb"], // something like this should work as expected
    "args": [
        "/c",
        "dotnet",
        "jb",
        "cleanupcode",
        "My.sln",
        "--profile=Teams: Clean up code",
        "--settings=My.sln.DotSettings",
        "--include=",
        "${staged}"
    ]
}
```

Return:

```
[Husky]  Preparing task 'Clean up code'
[Husky]  Executing task 'Clean up code' ...
JetBrains Cleanup Code 2023.3 RC 1
Running on x64 OS in x64 architecture, .NET 8.0.0 under Microsoft Windows 10.0.22631
Invalid option 'include'. Error: Requires value. Set of relative paths separated by ';' that defines which files should be cleaned up. Might contains single ('*') and double ('**') wildcards..
  ❌ Task 'Clean up code' failed in 1 083ms
husky - pre-commit hook exited with code 1 (error)
```

**Reply by alirezanet** [reply]
I remember I've used jetbrain tools in the past but didn't face this problem, tonight I'll try to find my configuration and share it with you.

**Reply by alirezanet** [reply]
Hi @AlaiyaNetwork,

I checked it out and it seems there's no easy way to pass staged files to the jb tool using a semicolon separator at the moment. but I'm planning to add some new features in the future that will make it easier and customizable. For now, you can try using a script to parse and convert the paths as a workaround. I'll keep you posted once the new features are ready to be released.
You can also follow this issue #99

**Comment by alirezanet** [comment, marked as answer]
Hi @AlaiyaNetwork,

Could you please take a look at version `v0.6.4` and let me know if it resolves your issue? In this release, you now have the ability to add a separator to the `${staged}` variable, for instance, `${staged:;}`.

I've included an example in the release notes, available [here](https://github.com/alirezanet/Husky.Net/releases/tag/v0.6.4).
```json
{
   "name": "jb cleanup",
   "group": "pre-commit",
   "command": "dotnet",
   "pathMode": "relative",
   "include": ["**/*.cs", "**/*.vb", "*.cs"],
   "args": [
      "jb",
      "cleanupcode",
      "--include=${staged:;}", 
      "solutionFile.sln"
   ]
}
```

Let me know if you have any questions!

**Reply by AlaiyaNetwork** [reply]
Good day! Sorry for the delay in responding. I have checked this version. Everything works fine! Thank you very much!
